examples/aco_tsp/run_tsp.py

```diff
@@ -0,0 +1,47 @@
+from collections import defaultdict
+
+import matplotlib.pyplot as plt
+from aco_tsp.model import AcoTspModel, TSPGraph
+
+
+def main():
+    # tsp_graph = TSPGraph.from_random(num_cities=20, seed=1)
+    tsp_graph = TSPGraph.from_tsp_file("aco_tsp/data/kroA100.tsp")
+    model_params = {
+        "num_agents": tsp_graph.num_cities,
+        "tsp_graph": tsp_graph,
+    }
+    number_of_episodes = 50
+
+    results = defaultdict(list)
+
+    best_path = None
+    best_distance = float("inf")
+
+    model = AcoTspModel(**model_params)
+
+    for e in range(number_of_episodes):
+        # model = AcoTspModel(**model_params)
+        model.step()
+        results["best_distance"].append(model.best_distance)
+        results["best_path"].append(model.best_path)
+        print(
+            f"Episode={e + 1}; Min. distance={model.best_distance}; pheromone_1_8={model.grid.G[17][15]['pheromone']}"
+        )
+        if model.best_distance < best_distance:
+            best_distance = model.best_distance
+            best_path = model.best_path
+            print(f"New best distance:  distance={best_distance}")
+
+    print(f"Best distance: {best_distance}")
+    print(f"Best path: {best_path}")
+    print(model.datacollector.get_model_vars_dataframe())
+
+    _, ax = plt.subplots()
+    ax.plot(results["best_distance"])
+    ax.set(xlabel="Episode", ylabel="Best distance", title="Best distance per episode")
```

Comment:
Somehow this doesn't work, the title and labels don't show up. Maybe some of the [other examples](https://github.com/search?q=repo%3Aprojectmesa%2Fmesa-examples%20title&type=code) can help (I see everyone does it differently).

Reply:
No, the cause is because the axis is initialized using the global non-thread-safe method, `plt.*`. It has to be done this way using the OOP API: https://github.com/projectmesa/mesa/blob/c96ee458ca480c786802587fa4e8d0d997b4de95/mesa/visualization/components/matplotlib.py#L11-L12.

Reply:
Oh, is this not the SolaraViz? Then my comment above is relevant.

Reply:
Sorry, it was in SolaraViz, but I placed the review comment on the wrong place.

Reply:
When using SolaraViz, I use the OOP API.  This comment is for the script that doesn't use Solara.  Either way, I see the axis labels as intended.  Which plots are unlabelled for you?